include/alpaka/core/ConcurrentExecPool.hpp

```diff
@@ -503,12 +503,14 @@ namespace alpaka::core
                 // Checks whether pool is being destroyed, if so, stop running (lazy check without mutex).
                 while(!m_bShutdownFlag)
                 {
-                    auto currentTaskPackage = std::shared_ptr<ITaskPkg>{nullptr};
-
-                    // Use popTask so we only ever have one reference to the ITaskPkg
-                    if(popTask(currentTaskPackage))
                     {
```

Comment:
please add a comment on why this code is in its own code block. I assume it has something to do with guaranteeing when ` currentTaskPackage` is destroyed. The comment should avoid that someone is removing the paranteeses.